lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -295,6 +283,11 @@ protected function validateRuntimeMetadata($class, $parent)
                         throw MappingException::invalidClassInDiscriminatorMap($subClass, $class->name);
                     }
                 }
+            } else {
+                assert($parent instanceof ClassMetadataInfo); // make static code analysis happy, see https://github.com/doctrine/orm/pull/8402#issuecomment-855805708
```

Comment:
Please create a Github issue instead :pray:

Reply:
Not sure I got you right... You mean remove this line here? Then PHPStan will complain.

I thought you asked me to add this code here to make PHPStan happy, and to open another issue/PR to remove it again at the same time as fixing the typehints/param docblocks?

Reply:
I mean keep this line here, but change the comment to point to an issue clearly describing what technical debt we have, not a thread buried in an unrelated pull request :wink: